src/app/components/PortraitVideoModal/index.tsx

```diff
@@ -0,0 +1,108 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/react';
+import { useState } from 'react';
+import MediaLoader from '#app/components/MediaLoader';
+import { PortraitClipMediaBlock } from '#app/components/MediaLoader/types';
+import { LeftChevron, RightChevron } from '../icons';
+import styles from './index.styles';
+
+export interface PortraitVideoModalProps {
+  items: {
+    id: string;
+    title: string;
+    versionId: string;
+    duration: string;
+    kind: string;
+    guidance: string | null;
+    territories: string[];
+    isEmbeddingAllowed: boolean;
+    images: {
+      url: string;
+      urlTemplate: string;
+    }[];
+  }[];
+  initialVideoIndex: number;
+  onClose: () => void;
+}
+
+const PortraitVideoModal = ({
+  items,
+  initialVideoIndex,
+  onClose,
+}: PortraitVideoModalProps) => {
+  const [currentVideoIndex, setCurrentIndex] = useState(initialVideoIndex);
+  const video = items[currentVideoIndex];
+
+  const block: PortraitClipMediaBlock = {
+    type: 'portraitClipMedia',
+    model: {
+      type: 'video',
+      images: video.images.map(img => ({
+        source: img.url,
+        urlTemplate: img.urlTemplate,
+      })),
+      video: {
+        id: video.id,
+        title: video.title,
+        version: {
+          id: video.versionId,
+          duration: video.duration,
+          kind: video.kind,
+          guidance: video.guidance,
+          territories: video.territories,
+        },
+        isEmbeddingAllowed: video.isEmbeddingAllowed,
+      },
+    },
+  };
+
+  const handlePrev = () => {
+    if (currentVideoIndex > 0) {
+      setCurrentIndex(i => i - 1);
+    }
+  };
+
+  const handleNext = () => {
+    if (currentVideoIndex < items.length - 1) {
+      setCurrentIndex(i => i + 1);
+    }
+  };
+
+  return (
+    <div css={styles.modalWrapper}>
```

Comment:
I'd recommend the [dialog](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/dialog) element here as it may help towards accessibility for the modal

Reply:
See https://github.com/bbc/simorgh/pull/12700/commits/3d753803a2621ee110f9644f321e46139d8aee6f